### PR TITLE
feat: add `Expr::as_binary_op`

### PR DESCRIPTION
**Pull request by asterite**
# Description

## Problem

Part of #5668

## Documentation

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [x] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: noir_stdlib/src/meta/op.nr [hunk 1]
```diff
@@ -19,3 +19,74 @@ impl UnaryOp {
...
     }
 }
+
+struct BinaryOp {
+    op: Field
+}
+
+impl BinaryOp {
+    fn is_add(self) -> bool {
+        self.op == 0
```
Comment (TomAFrench) [inline noir_stdlib/src/meta/op.nr:29]: Worth creating some globals for this for readability?

Comment (asterite) [reply noir_stdlib/src/meta/op.nr:29]: Hey Tom, welcome back! 👋 

I'm not sure. I think it's clear what the number means based on the enclosing method name. But if you think it's worth it, let me know and I'll do it.

**Review by jfecher — APPROVED** [review]
LGTM. I don't see us needing to use these constants elsewhere so I don't think we need globals for them.

File: noir_stdlib/src/meta/op.nr (hunk 1 above)
Comment (TomAFrench) [reply noir_stdlib/src/meta/op.nr:29]: I don't have overly strong opinions on this. Based on the number of these that we have, I would add these as a quasi-enum but it's 2-1 on this point.

**Pull request merged**